### PR TITLE
Added dart to firebase CLI

### PR DESCRIPTION
**Pull request by maneesht**
DO NOT MERGE UNTIL DART GENERATION IS READY
1. Added Dart Experiment
2. Added `enable_dart` flag
3. Added auto detection of Dart Platform

### CODE REVIEW

File: src/emulator/downloadableEmulators.ts [hunk 1]
```diff
@@ -57,9 +57,9 @@ const EMULATOR_UPDATE_DETAILS: { [s in DownloadableEmulators]: EmulatorUpdateDet
   dataconnect:
     process.platform === "darwin"
       ? {
-          version: "1.3.5",
```
Comment (joehan) [inline src/emulator/downloadableEmulators.ts:60]: This wont be relevant for long, but we should remove this for now. Will dart be included in the next normal emulator release, or is it hidden behind an experiment

**Review by joehan — APPROVED** [review]
Change LGTM, just need to clean up the downloadable emualtor changes

File: src/emulator/downloadableEmulators.ts (hunk 1 above)
Comment (maneesht) [reply src/emulator/downloadableEmulators.ts]: Removed these changes. Thanks for the find!

File: src/init/features/dataconnect/sdk.ts [hunk 2]
```diff
@@ -23,6 +24,7 @@ import { DataConnectEmulator } from "../../../emulator/dataconnectEmulator";
 import { FirebaseError } from "../../../error";
 import { camelCase, snakeCase, upperFirst } from "lodash";
 import { logSuccess, logBullet } from "../../../utils";
+import * as experiments from "../../../experiments";
```
Comment (joehan) [inline src/init/features/dataconnect/sdk.ts:27]: Unused?

File: src/commands/dataconnect-sdk-generate.ts [hunk 3]
```diff
@@ -7,6 +7,7 @@ import { needProjectId } from "../projectUtils";
 import { load } from "../dataconnect/load";
 import { readFirebaseJson } from "../dataconnect/fileUtils";
 import { logger } from "../logger";
+import * as experiments from "../experiments";
```
Comment (joehan) [inline src/commands/dataconnect-sdk-generate.ts:10]: Unused, remove.

**Review by joehan — APPROVED** [review]
LGTM after 2 small changes and adding it to schema/conect-yaml.json

**Pull request merged**